packages/build/src/error/types.ts

```diff
@@ -39,10 +113,44 @@ export const getTypeInfo = function ({ type }) {
 //    "All errors" bookmark depending on whether we should get notified on Slack
 //    for new errors of that type. You must use the bookmark menu action "Update
 //    with current filters"
-const TYPES = {
+export interface ErrorType {
+  title: TitleFunction | string
+  locationType?: string
+  showErrorProps?: boolean
+  rawStack?: boolean
+  showInBuildLog?: boolean
+  group?: GroupFunction
+  severity: keyof typeof ErrorSeverity
+  stackType: keyof typeof StackType
+}
+
+const ErrorTypeMap = {
+  cancelBuild: 'cancelBuild',
+  resolveConfig: 'resolveConfig',
+  dependencies: 'dependencies',
+  pluginInput: 'pluginInput',
+  pluginUnsupportedVersion: 'pluginUnsupportedVersion',
+  buildCommand: 'buildCommand',
+  functionsBundling: 'functionsBundling',
+  secretScanningFoundSecrets: 'secretScanningFoundSecrets',
+  failPlugin: 'failPlugin',
+  failBuild: 'failBuild',
+  pluginValidation: 'pluginValidation',
+  pluginInternal: 'pluginInternal',
+  ipc: 'ipc',
+  corePlugin: 'corePlugin',
+  coreStep: 'coreStep',
+  api: 'api',
+  exception: 'exception',
+  telemetry: 'telemetry',
+} as const
+
+type ErrorTypes = keyof typeof ErrorTypeMap
+
+const TYPES: { [T in ErrorTypes]: ErrorType } = {
```

Comment:
I'm guessing there's probably better ways to deal with this in the future if we break down this large object and create separate types for these 🤔